src/app/utilities/transmart-utilities/transmart-mapper.ts

```diff
@@ -255,4 +251,13 @@ export class TransmartMapper {
     return map;
   }
 
+  public static mapDimensions(transmartDimensions: TransmartDimension[]) {
+    let cohortTypes = [];
+    transmartDimensions.forEach( (transmartDimension: TransmartDimension) => {
+      let dimension = new Dimension(transmartDimension.name);
+      dimension.sortIndex = transmartDimension.sortIndex;
```

Comment:
why is the sortIndex assignment? what is it used for?

Reply:
this is to make sure dimensions in the dropdown are shown in a proper order. It is important because of the hierarchy between dimensions (biomaterial belongs to biosource, biosource -> diagnosis, diagnosis -> patient, image -> patient)
I just realized the sorting is not working correctly now, will fix it.

Reply:
how exactly does sortIndex work?

Reply:
it is just a number defining the position on the dimension list